src/edu/washington/escience/myria/CsvTupleWriter.java

```diff
@@ -57,9 +62,20 @@ public void writeColumnHeaders(final List<String> columnNames) throws IOExceptio
   public void writeTuples(final ReadableTable tuples) throws IOException {
     final String[] row = new String[tuples.numColumns()];
     /* Serialize every row into the output stream. */
+    Schema tbsc = tuples.getSchema();
     for (int i = 0; i < tuples.numTuples(); ++i) {
       for (int j = 0; j < tuples.numColumns(); ++j) {
-        row[j] = tuples.getObject(j, i).toString();
+        Type type = tbsc.getColumnType(j);
+        if (type.equals(Type.BLOB_TYPE)) {
+          // write the file out
+          // add filename to the csv file
+          String filename = UUID.randomUUID().toString();
```

Comment:
Usually we call a mature API to do something for us in a professional way. I'm just making these up but these are possible pros/cons:
- What if we want to put such a file in one directory? Or with a specific prefix?
- What if a file with the same UUID is already there?
All these are already handled by `File.createTempFile()`.

On the other hand, `File.createTempFile()` returns a `File` so you don't need to create a `File` in `writeBBtoFile` anymore.